src/vos/vos_dtx.c

```diff
@@ -133,14 +136,17 @@ dtx_rec_free(struct btr_instance *tins, struct btr_record *rec, void *args)
 {
 	umem_off_t	*umoff;
 
-	D_ASSERT(args != NULL);
 	D_ASSERT(!UMOFF_IS_NULL(rec->rec_off));
 
-	/* Return the record addreass (offset in SCM). The caller will
-	 * release it after using.
-	 */
-	umoff = (umem_off_t *)args;
-	*umoff = rec->rec_off;
+	if (args != NULL) {
+		/* Return the record addreass (offset in SCM).
+		 * The caller will release it after using.
+		 */
+		umoff = (umem_off_t *)args;
+		*umoff = rec->rec_off;
+	} else {
+		dtx_rec_release(&tins->ti_umm, rec->rec_off, true, true);
```

Comment:
will it cause the rec add to undo log two times? (umem_tx_add_ptr called both within dtx_rec_release and L151)

Reply:
That is possible, but is it harmful or not?

Reply:
yes, this is harmless